Lib/test/datetimetester.py

```diff
@@ -2676,6 +2676,14 @@ def test_fromisoformat_fails_datetime(self):
                 with self.assertRaises(ValueError):
                     self.theclass.fromisoformat(bad_str)
 
+    def test_fromisoformat_fails_surrogate(self):
+        # Test that when fromisoformat() fails with a surrogate character as
+        # the separator, the error message contains the original string
+        dtstr = "2018-01-03\ud80001:0113"
+
+        with self.assertRaisesRegex(ValueError, f".*{dtstr}"):
```

Comment:
IMO just checking that `ValueError` is raised is enough.

Reply:
The regex is actually the point of the test. Before these changes, the error message was accidentally including the *sanitized* string, so `2018-01-03\ud80001:0113` would give you something like `Invalid isoformat str: 2018-01-03T01:0113`.

As a result of this test, I also realized that the C and pure Python versions had slightly different error messages, so I corrected that (and this test enforces it).

If we don't care to enforce any particular conditions on the error message, we can drop the whole test, since the part about raising `ValueError` has pretty good test coverage elsewhere.